src/java.base/share/classes/java/lang/invoke/LambdaForm.java

```diff
@@ -1393,11 +1393,11 @@ private Name(Name that, Object constraint) {
         Name(BasicType type) { this(-1, type); }
 
         BasicType type() { return type; }
-        int index() { return index; }
+        int index() { return offsetIndex - 1; }
         boolean initIndex(int i) {
-            if (index != i) {
-                if (index != -1)  return false;
-                index = (short)i;
+            if (offsetIndex != i + 1) {
+                if (offsetIndex != 0)  return false;
+                offsetIndex = (short) (i + 1);
```

Comment:
Can we pull `(short) (i + 1);` into a local variable, given that we need it on all paths?